### PR TITLE
Fix executing gcode inside long comments

### PR DESCRIPTION
**Pull request by milosfec**
When line was langer than 96 chars, remaining part of the line was
parsed as new line.
If the line included long comment and after 96 chars there was a
gcode, that gcode was executed.

Throw away rest of the line to avoid the issue.

Solves BFW-1264

### CODE REVIEW

File: src/common/media.cpp [hunk 1]
```diff
@@ -272,6 +274,11 @@ void media_loop(void) {
                     }
                     break;
                 }
+                while (len == MAX_CMD_SIZE && buffer[MAX_CMD_SIZE - 1] != '\n') {
+                    // Read until the end of the line
+                    f_gets(buffer, sizeof(buffer), &media_print_fil);
```
Comment (dragomirecky) [inline src/common/media.cpp:279]: /nitpick/ In the case that this `f_gets` call starts to fail, we might end up in an infinite cycle. Right?

Comment (milosfec) [reply src/common/media.cpp:279]: Good point, thanks

File: src/common/media.cpp [hunk 2]
```diff
@@ -272,6 +275,13 @@ void media_loop(void) {
                     }
                     break;
                 }
+                while (len == MAX_CMD_SIZE && buffer[MAX_CMD_SIZE - 1] != '\n') {
+                    // Read until the end of the line
+                    if (!f_gets(buffer, sizeof(buffer), &media_print_fil)) {
```
Comment (DRracer) [inline src/common/media.cpp:280]: And what will be the state of `media_current_position` when the line is too long? Could you please add some unit tests so that we can verify these potential edge cases?

File: src/common/media.cpp [hunk 3]
```diff
@@ -272,6 +275,13 @@ void media_loop(void) {
...
+                        goto no_data;
```
Comment (espr14) [inline src/common/media.cpp:281]: I know this is already an if-hell suitable for refactor. However, I would avoid `goto`. You can create `media_no_data()`. Then you'd place here:
```
media_no_data();
return;
```
and this instead of `no_data:` block:
```
} else {
    media_no_data();
    return;
}
```

**Comment by milosfec** [comment]
Replaced by https://github.com/prusa3d/Prusa-Firmware-Buddy/pull/1372
